--- a/node_build/make.js
+++ b/node_build/make.js
@@ -25,7 +25,8 @@ var TestRunner = require('./TestRunner')
 const CjdnsTest = require('./CjdnsTest');
 const GetVersion = require('./GetVersion');
 
-var CFLAGS = process.env['CFLAGS'];
+var CFLAGS = process.env['CFLAGS'].trim().replace('  ',' ');
+process.env['CFLAGS']=CFLAGS;
 var LDFLAGS = process.env['LDFLAGS'];
 var NO_MARCH_FLAG = ['arm', 'ppc', 'ppc64'];
 
@@ -300,6 +301,7 @@ Builder.configure({
         builder.config.libs.push(libuvLib);
         if (!android) {
             builder.config.libs.push('-lpthread');
+            builder.config.libs.push('-lssp');
         }
 
         if (builder.config.systemName === 'win32') {
