--- a/sysdep/linux/netlink.c
+++ b/sysdep/linux/netlink.c
@@ -13,6 +13,11 @@
 #include <sys/uio.h>
 #include <errno.h>
 
+/* RTAX_CC_ALGO since Linux 4.0 */
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
 #undef LOCAL_DEBUG
 
 #include "nest/bird.h"
@@ -499,10 +504,12 @@ static inline ip_addr rta_get_ipa(struct
 static inline ip_addr rta_get_via(struct rtattr *a)
 {
   struct rtvia *v = RTA_DATA(a);
+#ifdef HAVE_MPLS_KERNEL
   switch(v->rtvia_family) {
     case AF_INET:  return ipa_from_ip4(ip4_ntoh(*(ip4_addr *) v->rtvia_addr));
     case AF_INET6: return ipa_from_ip6(ip6_ntoh(*(ip6_addr *) v->rtvia_addr));
   }
+#endif
   return IPA_NONE;
 }
 
@@ -836,7 +843,9 @@ err:
   return NULL;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
 STATIC_ASSERT(EA_KRT_METRICS + RTAX_CC_ALGO == EA_KRT_CONGCTL);
+#endif /* RTAX_CC_ALGO since Linux 4.0 */
 
 static void
 nl_add_metrics(struct nlmsghdr *h, uint bufsize, u32 *metrics, const char *cc_algo, int max)
@@ -846,9 +855,11 @@ nl_add_metrics(struct nlmsghdr *h, uint
 
   for (t = 1; t < max; t++)
     if (metrics[0] & (1 << t))
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
       if (t == RTAX_CC_ALGO)
         nl_add_attr_str(h, bufsize, t, cc_algo);
       else
+#endif /* RTAX_CC_ALGO since Linux 4.0 */
         nl_add_attr_u32(h, bufsize, t, metrics[t]);
 
   nl_close_attr(h, a);
@@ -869,6 +880,7 @@ nl_parse_metrics(struct rtattr *hdr, u32
     if (a->rta_type >= max)
       continue;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
     if (a->rta_type == RTAX_CC_ALGO)
     {
       *cc_algo = rta_get_str(a);
@@ -881,6 +893,7 @@ nl_parse_metrics(struct rtattr *hdr, u32
       metrics[a->rta_type] = 0;
     }
     else
+#endif /* RTAX_CC_ALGO since Linux 4.0 */
     {
       if (RTA_PAYLOAD(a) != 4)
 	return -1;
@@ -1424,9 +1437,11 @@ nl_send_route(struct krt_proto *p, rte *
     int id = ea->id - EA_KRT_METRICS;
     metrics[0] |= 1 << id;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
     if (id == RTAX_CC_ALGO)
       cc_algo = ea->u.ptr->data;
     else
+#endif /* RTAX_CC_ALGO since Linux 4.0 */
       metrics[id] = ea->u.data;
   }
 
@@ -1874,6 +1889,7 @@ nl_parse_route(struct nl_parse_state *s,
 		.u.data = metrics[t],
 	      };
 	    }
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
           else if (t == RTAX_CC_ALGO)
 	    {
 	      struct adata *ad = lp_alloc_adata(s->pool, strlen(cc_algo));
@@ -1885,6 +1901,7 @@ nl_parse_route(struct nl_parse_state *s,
 		.u.ptr = ad,
 	      };
 	    }
+#endif /* RTAX_CC_ALGO since Linux 4.0 */
 	  else
 	    {
 	      ea->attrs[n++] = (eattr) {
